### PR TITLE
HI individual PQ link for HH cases without individual case ID

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Motivation and Context
Some services need to be able to spawn HI child cases on PQ links without specifying an individual case ID, we can just generate a new random ID.

# What has changed
* Generate HI case ID if not present on PQ link event for individual QID/HH case

# How to test?
Build and run linked AT's (WIP)

# Links
https://trello.com/c/6peilnCZ/988-change-for-hi-case-from-pq-link-5
https://github.com/ONSdigital/census-rm-acceptance-tests/pull/271

### CODE REVIEW

**Comment by AdamHawtin** [comment]
[Change for HI case from PQ link (5)](https://trello.com/c/6peilnCZ/988-change-for-hi-case-from-pq-link-5)

File: src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java [hunk 1]
```diff
@@ -110,11 +109,8 @@ private void logUnlinkedCaseEventAgainstOriginalCaseIfQidIsNowLinkedToNewCase(
     }
   }
 
-  private boolean checkRequiredFieldsForIndividualHI(
-      String questionnaireId, String caseType, String individualCaseId) {
-    return (isIndividualQuestionnaireType(questionnaireId)
-        && caseType.equals("HH")
-        && individualCaseId != null);
+  private boolean checkRequiredFieldsForIndividualHI(String questionnaireId, String caseType) {
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java:112]: This function is looking pretty pointless

Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java:112]: although I suppose it's kinda 'documentation'

File: src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java [hunk 2]
```diff
@@ -110,11 +109,8 @@ private void logUnlinkedCaseEventAgainstOriginalCaseIfQidIsNowLinkedToNewCase(
...
+    return (isIndividualQuestionnaireType(questionnaireId) && caseType.equals("HH"));
   }
 
   private void throwIllegalArgumentExceptionIfIndQIDAndIndCaseIdPresentButCaseTypeNotHH(
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java:116]: While you're tidying things up, you could rename this ridiculous function

Comment (AdamHawtin) [reply src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java]: @NickGrantONS Renamed it to something a little less extravagantly long (but hopefully still descriptive enough).

File: src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java [hunk 3]
```diff
@@ -52,19 +51,18 @@ public void processQuestionnaireLinked(
 
     Case caze = caseService.getCaseByCaseId(UUID.fromString(uac.getCaseId()));
 
-    if (checkRequiredFieldsForIndividualHI(
-        questionnaireId, caze.getCaseType(), uac.getIndividualCaseId())) {
-      caze =
-          caseService.prepareIndividualResponseCaseFromParentCase(
-              caze, UUID.fromString(uac.getIndividualCaseId()));
+    throwExceptionIfArgumentsInvalid(caze, questionnaireId, uac.getIndividualCaseId());
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java:54]: I'm not a big fan of `throwException` type stuff... it smacks of test code. I would call it `checkArgumentsValid`

**Pull request merged**